Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/ComponentAdapter.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/UI/DocumentPropertyEditor/ComponentAdapter.h>
+#include <QtCore/QTimer>
+
+namespace AZ::DocumentPropertyEditor
+{
+    ComponentAdapter::ComponentAdapter()
+    {
+    }
+
+    ComponentAdapter::ComponentAdapter(AZ::Component* componentInstace)
+    {
+        SetComponent(componentInstace);
+    }
+
+    ComponentAdapter::~ComponentAdapter() = default;
+
+    void ComponentAdapter::OnEntityComponentPropertyChanged(AZ::ComponentId componentId)
+    {
+        if (m_componentInstance->GetId() == componentId)
+        {
+            m_refreshQueue++;
+            QTimer::singleShot(
+                0,
+                [this]()
+                {
+                    DoRefresh();
+                });
+        }
+    }
+
+    void ComponentAdapter::InvalidatePropertyDisplay([[maybe_unused]] AzToolsFramework::PropertyModificationRefreshLevel level)
+    {
+        m_refreshQueue++;
```

Comment:
If later we're going to be able to actually implement the various refresh levels, I would go ahead and store it now instead of doing a counter, something like:

```
if (level > m_queuedRefreshLevel)
{
    m_queuedRefreshLevel = level;
}
```

And then later in the `DoRefresh` it could look like:

```
void ComponentAdapter::DoRefresh()
    {
        if (m_queuedRefreshLevel == Refresh_None)
        {
            return;
        }

        NotifyResetDocument();

        m_queuedRefreshLevel = Refresh_None;
    }
```

Reply:
Thanks, I've stored the refresh levels instead of doing a counter in case we want to use them in the future :)